its/src/test/java/com/sonar/maven/it/suite/JavaTest.java

```diff
@@ -120,6 +121,31 @@ void setJavaVersionProperties() throws IOException {
       entry("sonar.java.target", "1.8"));
   }
 
+  @Test
+  void whenPropertyJavaHomeExistsThenItIsUsedAsDefaultValue() throws IOException {
+    //verify that property java home exists, it may not be pointing to a jdk
+    if (!System.getProperties().containsKey("java.home")
+      && !System.getenv().containsKey("JAVA_HOME")) {
+      return;
+    }
+
+
+    File outputProps = temp.resolve("out.properties").toFile();
+    outputProps.createNewFile();
+    File pom = ItUtils.locateProjectPom("jdkHome/extractFromJre");
+
+    MavenBuild build = MavenBuild.create(pom)
+      .setGoals(sonarGoal())
+      .setProperty("sonar.scanner.internal.dumpToFile", outputProps.getAbsolutePath());
+    executeBuildAndValidateWithoutCE(build);
+
+    Properties props = getProps(outputProps);
+    String jdkHome = props.getProperty("sonar.java.jdkHome");
+    //the jdk is not configured in the project, so default value should be used
+    assertThat(jdkHome).isIn(Arrays.asList(System.getProperty("java.home"), System.getenv().get("JAVA_HOME")));
```

Comment:
Can we tamper with these values from outside of the test? Shouldn't we check none of these values can be empty?

Reply:
I think we should test for wrong user-configured values and should not try to recover in that case.